llama-http/src/inference.rs

```diff
@@ -0,0 +1,105 @@
+use llama_rs::{InferenceParameters, InferenceSnapshot};
+use rand::thread_rng;
+use std::convert::Infallible;
+
+use crate::cli_args::CLI_ARGS;
+use flume::{unbounded, Receiver, Sender};
+
+#[derive(Debug)]
+pub struct InferenceRequest {
+    /// The channel to send the tokens to.
+    pub tx_tokens: Sender<Result<String, hyper::Error>>,
+
+    pub num_predict: Option<usize>,
+    pub prompt: String,
+    pub n_batch: Option<usize>,
+    pub top_k: Option<usize>,
+    pub top_p: Option<f32>,
+    pub repeat_penalty: Option<f32>,
+    pub temp: Option<f32>,
+}
+
+pub fn initialize_model_and_handle_inferences() -> Sender<InferenceRequest> {
+    // Create a channel for InferenceRequests and spawn a thread to handle them
+
+    let (tx, rx) = unbounded();
+
+    std::thread::spawn(move || {
+        let args = &*CLI_ARGS;
+
+        // TODO Preload prompt
+
+        // Load model
+        let (mut model, vocabulary) =
+            llama_rs::Model::load(&args.model_path, args.num_ctx_tokens as i32, |_progress| {
+                println!("Loading model...");
+            })
```

Comment:
Would be good if you use the same code as in main.rs, which shows an indication of progress:

```
[2023-03-20T19:05:01Z INFO  llama_cli] Loaded tensor 8/291
[2023-03-20T19:05:01Z INFO  llama_cli] Loaded tensor 16/291
[2023-03-20T19:05:02Z INFO  llama_cli] Loaded tensor 24/291
[2023-03-20T19:05:03Z INFO  llama_cli] Loaded tensor 32/291
```

Using the `log` crate for the server sounds like it would also be a good idea. It's even more important for a server to have timestamped logs and offer configurable output levels.

Reply:
Agreed, added in 541b498 and 7b52b83